master/internal/api_runs.go

```diff
@@ -503,3 +504,135 @@ func (a *apiServer) KillRuns(ctx context.Context, req *apiv1.KillRunsRequest,
 	}
 	return &apiv1.KillRunsResponse{Results: results}, nil
 }
+
+func (a *apiServer) DeleteRuns(ctx context.Context, req *apiv1.DeleteRunsRequest,
+) (*apiv1.DeleteRunsResponse, error) {
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+	// Get active runs and kill
+	var activeRuns []int32
+	activeStates := []model.State{model.ActiveState, model.PausedState, model.RunningState}
+
+	killQuery := db.Bun().NewSelect().
+		ModelTableExpr("runs AS r").
+		Model(&activeRuns).
+		Column("r.id").
+		Join("LEFT JOIN trials_v2 t ON r.id=t.run_id").
+		Join("LEFT JOIN experiments e ON r.experiment_id=e.id").
+		Join("JOIN projects p ON r.project_id = p.id").
+		Join("JOIN workspaces w ON p.workspace_id = w.id").
+		Where("r.state IN (?)", bun.In(activeStates)).
```

Comment:
Isn't this going to race with run completion? What happens if a run finishes after this query but before `a.KillRuns` is called? Should `a.KillRuns` just be able to take the entire list of runs and not care if it can't find one? Philosophically if we do that, then it is idempotent. I think we should try to make all APIs idempotent because it's a useful property to build on top of.

Reply:
We are no longer going to kill runs on deletion. Only deleting runs in terminal state